### PR TITLE
[WIP] THREESCALE-11281 - bump to go 1.22

### PR DESCRIPTION
**Pull request by valerymo**
Jira: https://issues.redhat.com/browse/THREESCALE-11281

## Verification
1. Install and sanity validation
3scale installed locally

```
cd 3scale-operator
make install
export NAMESPACE=3scale-test
oc new-project $NAMESPACE
oc project $NAMESPACE
make download
make run
```
Installed correctly
routes/UI verified

```
~/go/3scale-operator oc project
Using project "3scale-test" on server "https://api.vmo01.xxxxx".
~/go/3scale-operator oc get deploy
NAME                 READY   UP-TO-DATE   AVAILABLE   AGE
apicast-production   1/1     1            1           43h
apicast-staging      1/1     1            1           43h
backend-cron         1/1     1            1           43h
backend-listener     1/1     1            1           43h
backend-redis        1/1     1            1           43h
backend-worker       1/1     1            1           43h
system-app           1/1     1            1           43h
system-memcache      1/1     1            1           43h
system-postgresql    1/1     1            1           43h
system-redis         1/1     1            1           43h
system-searchd       1/1     1            1           43h
system-sidekiq       1/1     1            1           43h
zync                 1/1     1            1           43h
zync-database        1/1     1            1           43h
zync-que             1/1     1            1           43h
```

2. CVEs analysys
Quay.io Security Scan - Passed

- Image was built to see CVE check in Quay.io. Security Scan is running / will be updated
- Repo: quay.io/vmogilev_rhmi/3scale-operator:test-go121
- Command used to build image:
```
docker build . --platform linux/amd64 -t quay.io/vmogilev_rhmi/3scale-operator:test-go121
```

image

### CODE REVIEW

**Review by carlkyrillos — COMMENTED** [review]
Reviewed the changed files and verified the operator works when run locally, OLM(cluster-scoped), and OLM(namespace-scoped).

This is ready to merge once 2.15 is released.

/lgtm

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.20.14 as builder
+FROM golang:1.21.13 as builder
```
[the remark this answers is not in the document]
Comment (valerymo) [reply Dockerfile:2]: @carlkyrillos , sorry for delay with reply. I used scopeo tool:
$skopeo list-tags docker://docker.io/library/golang |grep 1.21 
1.21.13 - was latest

[the remark this answers is not in the document]
Comment (carlkyrillos) [reply Dockerfile:2]: Sounds good thanks

**Pull request closed without merging**